### PR TITLE
Adds bigip_software_update module

### PR DESCRIPTION
**Pull request by caphrim007**
##### SUMMARY

This module can be used to adjust the ways by which a bigip communicates
back to F5 to check for system updates.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

bigip_software_update

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /here/test/integration/ansible.cfg
  configured module search path = [u'/here/library']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Dec 12 2017, 16:55:09) [GCC 4.9.2]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/network/f5/bigip_software_update.py [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+short_description: Manage the software update settings of a BIG-IP
+description:
+  - Manage the software update settings of a BIG-IP.
+version_added: "2.5"
+options:
+  auto_check:
+    description:
+      - Specifies whether to automatically check for updates on the F5
+        Networks downloads server.
+    choices:
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_software_update.py]: FYI you can replace this with `type: bool`

File: lib/ansible/modules/network/f5/bigip_software_update.py [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+      - no
+  frequency:
+    description:
+      - Specifies the schedule for the automatic update check.
+    choices:
+      - daily
+      - monthly
+      - weekly
+notes:
+  - Requires the f5-sdk Python package on the host This is as easy as pip
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_software_update.py]: `C(pip install f5-sdk)`

**Review by gundalow — APPROVED** [review]
To comments, though they don't block you merging

File: lib/ansible/modules/network/f5/bigip_software_update.py (hunk 2 above)
Comment (caphrim007) [reply lib/ansible/modules/network/f5/bigip_software_update.py]: should i change this across all of the modules?

Comment (gundalow) [reply lib/ansible/modules/network/f5/bigip_software_update.py]: It's a minor issue. Wonder if it can go in the docs fragment

Comment (caphrim007) [reply lib/ansible/modules/network/f5/bigip_software_update.py]: hmm, thats a good point. i havent been much concerned with optimizations lately, but that would surely be one.

Notes in doc fragments are presumably merged with notes in modules?

**Pull request merged**